src/core/server/saved_objects/migrations/core/migrate_raw_docs.ts

```diff
@@ -111,10 +112,7 @@ export function migrateRawDocsSafely({
     const corruptSavedObjectIds: string[] = [];
     const options = { namespaceTreatment: 'lax' as const };
     for (const raw of rawDocs) {
-      // Do not transform documents of unknown types
-      if (raw?._source?.type && !knownTypes.has(raw._source.type)) {
-        processedDocs.push(raw);
-      } else if (serializer.isRawSavedObject(raw, options)) {
```

Comment:
This part of the revert was not strictly necessary, but it felt safer to get back to the behavior where migrating unknown doc fails, to be consistent with our strategy to fail the migration in such case (even if going that far should never occur)